lib/Textmaster/Model/Project.php

```diff
@@ -219,7 +221,46 @@ public function createDocument()
             throw new BadMethodCallException('The project must be saved before adding documents.');
         }
 
-        return new Document($this->client, array('project_id' => $this->getId()));
+        $document = new Document($this->client, array('project_id' => $this->getId()));
+        $document->setProject($this);
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addDocuments(array $documents)
```

Comment:
why not making 2 private methods ? One to generate the data to send through the API. And another to dispatch events .